### ISSUE TITLE
report.txt not generated when no output file is specified

### ISSUE THREAD

**Issue by rnjudge** [tern-tools/tern#731]
**Describe the bug**
The README tells the user to run `tern report -i debian:jessie` and `Take a look at report.txt to see what packages are installed in the Docker image and how Tern got this information.` Tern does not create report.txt, however, and instead prints this report information to the console. Additionally, `tern report --help` states:

```
 -o FILE, --output-file FILE
                        Write the report to a file. If no file is given the
                        default file in utils/constants.py will be used
```

but the default file (report.txt) is not used.  This is because of the following code in `report.py` which only writes the report to a file if an output file is specified, otherwise it prints it to the console. If we want to follow the current documentation guidelines, the `report_out` function should call write_report regardless if the `args.output_file` value exists or not.

```
151 def report_out(args, *images):                                                  
152     report = generate_report(args, *images)                                     
153     if not report:                                                              
154         logger.error("%s not a recognized plugin.", args.report_format)         
155     elif args.output_file:                                                      
156         write_report(report, args)                                              
157     else:                                                                       
158         print(report)  
```

If we don't want Tern to create a default report file (report.txt) when no output file is specified, we need to update our documentation accordingly. @nishakm do you have much of a preference either way? Personally, I would like to see a default report file generated instead of printed to the console. 

**To Reproduce**
1. run `tern report -i debian:jessie`
2. report.txt is not generated.

**Comment by nishakm** [comment]
@rnjudge Totally forgot that the default is supposed to be console out not write to a file. This is especially true for the use case of running tern in a container. Can you provide the situation where the console out use case breaks?

**Comment by rnjudge** [comment]
If that is the case then we need to update our documentation and command line directions as it currently says that `report.txt` will be generated when no output file is specified. Nowhere in the documentation does it mention printing the output file to console if an output file is not provided. This is easy to update but I opened the issue so that either way we chose, we were being consistent. Prior to merging PR #732, Tern was either printing to console _or_ putting the report in a user-specified output file. In the case of the html report, printing the huge html file to console doesn't seem to help anyone but I understand that making sense for the other formats.

**Comment by rnjudge** [comment]
For example, here is the command line directions for specifying an output file:
```
  -o FILE, --output-file FILE
                        Write the report to a file. If no file is given the
                        default file in utils/constants.py will be used

```

**Comment by nishakm** [comment]
How about updating the README.md to reflect the expected usage for the HTML format (i.e. use -o) ?

**Comment by nishakm** [comment]
> For example, here is the command line directions for specifying an output file:
> 
> ```
>   -o FILE, --output-file FILE
>                         Write the report to a file. If no file is given the
>                         default file in utils/constants.py will be used
> ```

@rnjudge Agree this is misleading and needs to be updated.

**Comment by rnjudge** [comment]
> How about updating the README.md to reflect the expected usage for the HTML format (i.e. use -o) ?

That's fine. Since we need to fix more than just reverting #732 (the logic before the merge is incorrect and would require further editing after a revert) I propose merging the html report and I can open a new PR to fix the documentation and the logic surrounding where the output goes?

**Comment by nishakm** [comment]
> > How about updating the README.md to reflect the expected usage for the HTML format (i.e. use -o) ?
> 
> That's fine. Since we need to fix more than just reverting #732 (the logic before the merge is incorrect and would require further editing after a revert) I propose merging the html report and I can open a new PR to fix the documentation and the logic surrounding where the output goes?

Sounds good to me. I will make the edit my comment in #740. It was all the concern I had.
